### PR TITLE
Algorithm history uses load version for "Load"

### LINKED ISSUE #33076: Error Message While Generating History as Script if Load is present
**Describe the bug**

An error message is produced when generating a scipt from the workspace history where Load was used in the history

**To Reproduce**

* Start workbench
* Load `SXD23767` from the training data
* Once loaded go to the workspace and show the history
* Click `Script to Clipboard`
* => The error will appear in the logger

The history does look okay...

**Expected behavior**

No error message should be generated.

**Screenshots**

Snapshot of text from logger

```
Load started
Load successful, Duration 4.27 seconds
Could not create a fresh version of Load version 1
Could not create a fresh version of Load version 1
```

**Platform/Version (please complete the following information):**
 - OS: macOS
 - OS Version: 11.6
 - Mantid Version: Development

### PR DESCRIPTION
**Pull request by AnthonyLim23**
**Description of work.**
After loading some data sets into Mantid and trying to generate a history (either by specifically asking for it or project recovery) an error message would appear in the log `Could not create a fresh version of Load version 1`. This was caused by the algorithm creation using the wrong version (it was the version of Load being called and not of the actual loader).
-->

**To test:**

Open Mantid
Load `SXD23767`
Get the history - no warnings should appear in the log
Paste the history into a script
Clear the ADS
Run the script (to check it works)

Fixes #33076 . 

This does not require release notes because its a bug that I added a couple of weeks ago.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/API/src/ScriptBuilder.cpp [hunk 1]
```diff
@@ -177,8 +177,8 @@ const std::string ScriptBuilder::buildAlgorithmString(const AlgorithmHistory &al
     Mantid::API::Algorithm_sptr algFresh;
     // create a fresh version of the algorithm - unmanaged
     if (name == "Load") {
-      algFresh =
-          AlgorithmManager::Instance().createUnmanaged(algHistory.getPropertyValue("LoaderName"), algHistory.version());
+      int version = atoi(algHistory.getPropertyValue("LoaderVersion").c_str());
```
Comment (martyngigg) [inline Framework/API/src/ScriptBuilder.cpp:180]: Can we use the [C++ version](https://en.cppreference.com/w/cpp/string/basic_string/stol) of this instead as it just accepts a `std::string`:
```
const int version = std::stoi(algHistory.getPropertyValue("LoaderVersion"))
```

File: Framework/API/test/ScriptBuilderTest.h [hunk 2]
```diff
@@ -573,6 +573,26 @@ class ScriptBuilderTest : public CxxTest::TestSuite {
     m_ads.remove("MUSR00022725");
   }
 
+  void test_Build_Load_Uses_Correct_version() {
```
Comment (martyngigg) [inline Framework/API/test/ScriptBuilderTest.h:576]: Thanks for adding a test 👍🏻

**Review by martyngigg — CHANGES_REQUESTED** [review]

File: Framework/API/src/ScriptBuilder.cpp [hunk 3]
```diff
@@ -177,8 +177,8 @@ const std::string ScriptBuilder::buildAlgorithmString(const AlgorithmHistory &al
     Mantid::API::Algorithm_sptr algFresh;
     // create a fresh version of the algorithm - unmanaged
     if (name == "Load") {
-      algFresh =
-          AlgorithmManager::Instance().createUnmanaged(algHistory.getPropertyValue("LoaderName"), algHistory.version());
+      int version = std::stoi(algHistory.getPropertyValue("LoaderVersion").c_str());
```
Comment (AnthonyLim23) [inline Framework/API/src/ScriptBuilder.cpp:180]:
```suggestion
      int version = std::stoi(algHistory.getPropertyValue("LoaderVersion"));
```

**Review by martyngigg — APPROVED** [review]
This looks good now. Confirmed that the error message no longer appears when creating the script from that has a loader where version 1 does not exist.

**Pull request merged**